yarn-project/end-to-end/src/e2e_key_registry.test.ts

```diff
@@ -181,84 +183,321 @@ describe('SharedMutablePrivateGetter', () => {
+    });
 
-      await wallets[0]
-        .setPublicAuthWit({ caller: wallets[1].getCompleteAddress().address, action }, true)
+    it('should succeed because we register our recipient manually and the lib checks our pxe', async () => {
+      // FIX THIS (#5834)
+      await pxe.registerRecipient(CompleteAddress.create(accountAddedToRegistry, Point.ZERO, partialAddress), [
+        masterNullifierPublicKey,
+        masterIncomingViewingPublicKey,
+        masterOutgoingViewingPublicKey,
+        masterTaggingPublicKey,
+      ]);
+      await testContract.methods
+        .test_nullifier_key_freshness(accountAddedToRegistry, masterNullifierPublicKey)
         .send()
         .wait();
+    });
 
-      await action.send().wait();
+    it('should fail when we put in a different key', async () => {
+      await expect(
+        testContract.methods
+          .test_nullifier_key_freshness(accountAddedToRegistry, masterIncomingViewingPublicKey)
+          .send()
+          .wait(),
+      ).rejects.toThrow(`Cannot satisfy constraint 'keys[key_type].eq(key)'`);
+    });
+
+    it('Now we add it to registry', async () => {
+      await keyRegistry
+        .withWallet(wallets[0])
+        .methods.register(
+          AztecAddress.fromField(accountAddedToRegistry),
+          partialAddress,
+          masterNullifierPublicKey,
+          masterIncomingViewingPublicKey,
+          masterOutgoingViewingPublicKey,
+          masterTaggingPublicKey,
+        )
+        .send()
+        .wait();
     });
 
-    it("checks our registry contract from test contract and finds our old public key because the key rotation hasn't been applied yet", async () => {
+    it('we start the change in the registry, it has not been applied yet, but we still see we have a fresh key due to it being added in the pxe', async () => {
       const { txHash } = await testContract.methods
-        .test_shared_mutable_private_getter_for_registry_contract(keyRegistry.address, 1, wallets[0].getAddress())
+        .test_shared_mutable_private_getter_for_registry_contract(keyRegistry.address, 1, accountAddedToRegistry)
         .send()
         .wait();
 
       const rawLogs = await pxe.getUnencryptedLogs({ txHash });
-      expect(Fr.fromBuffer(rawLogs.logs[0].log.data)).toEqual(new Fr(910));
+      expect(Fr.fromBuffer(rawLogs.logs[0].log.data)).toEqual(Fr.ZERO);
+
+      // Checks freshness of newly added keys, but the change hasn't been affected yet, but we have manually added it to our pxe so it should pass
+      await testContract.methods
+        .test_nullifier_key_freshness(accountAddedToRegistry, masterNullifierPublicKey)
+        .send()
+        .wait();
     });
 
-    it('checks our registry contract from test contract and finds the new nullifier public key that has been rotated', async () => {
+    it('in the case where the key exists both in the pxe and our registry, we know nothing weird will happen', async () => {
```

Comment:
"nothing weird will happen" is not descriptive enough. Also I feel all the test names are not concise enough.